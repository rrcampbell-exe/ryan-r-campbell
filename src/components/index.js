import BlogTags from './Blog/BlogTags'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Contact from './Contact/Contact'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'
import Loading from './Loading/Loading'
import Quote from './Quote/Quote'
import TextAndImage from './TextAndImage/TextAndImage'
import HamburgerMenu from './HamburgerMenu/HamburgerMenu'
import PopText from './PopText/PopText'
import PodcastTeaser from './Podcast/PodcastTeaser'
import Boasts from './Tech/Boasts'
import SkillsCloud from './Tech/SkillsCloud'
import BlogFooter from './Blog/BlogFooter'
import SubstackRedirect from './SubstackRedirect/SubstackRedirect'
import { CodeSpan, CodeBlock } from './CodeBlock/CodeBlock'

export { BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner, Loading, Quote, TextAndImage, BlogTags, HamburgerMenu, PopText, PodcastTeaser, Boasts, SkillsCloud, BlogFooter, CodeSpan, CodeBlock, SubstackRedirect }
