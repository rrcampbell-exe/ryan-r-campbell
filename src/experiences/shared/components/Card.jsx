import React from 'react'
import styled from 'styled-components'

// TODO: size should flex based on number of cards
// TODO: need to set up theme colors

const StyledDiv = styled.div`
  height: 16rem;
  width: 16rem;
  border: 1px solid #242424;
  border-radius: 4px;
  padding: 1rem;
  margin: 1rem;
`

const Card = ({ children }) => {
  return (
    <StyledDiv>
      {children}
    </StyledDiv>
  )
}

export default Card
