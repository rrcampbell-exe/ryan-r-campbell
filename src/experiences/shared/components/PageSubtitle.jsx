import React from 'react'
import styled from 'styled-components'

const StyledH2 = styled.h2`
  margin: 0 auto;
  padding: 0;
`

const PageSubtitle = ({ subtitle }) => {
  return (
    <StyledH2>
      {subtitle}
    </StyledH2>
  )
}

export default PageSubtitle
