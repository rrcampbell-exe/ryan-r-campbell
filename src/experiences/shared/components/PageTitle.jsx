import React from 'react'
import styled from 'styled-components'

const StyledH1 = styled.h1`
  margin: 0 auto;
  padding: 0;
`

const PageTitle = ({ title }) => {
  return (
    <StyledH1>
      {title}
    </StyledH1>
  )
}

export default PageTitle
