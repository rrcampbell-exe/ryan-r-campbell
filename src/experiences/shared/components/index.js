import BlogPost from './Blog/BlogPost'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Contact from './Contact/Contact'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'

export { BlogPost, BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner }
