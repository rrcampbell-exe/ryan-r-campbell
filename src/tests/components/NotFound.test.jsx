import { describe, expect, test } from 'vitest'
import { render } from '@testing-library/react'
import ContextAndRouterProvider from '../test-utils'
import { NotFound } from '../../components'

describe('NotFound', () => {
  test('default snapshot', () => {
    const notFound = render(
      <ContextAndRouterProvider>
        <NotFound />
      </ContextAndRouterProvider>
    )
    expect(notFound).toMatchSnapshot()
  })
  test('should render page not found', () => {
    const { getByText } = render(
      <ContextAndRouterProvider>
        <NotFound />
      </ContextAndRouterProvider>
    )

    const notFound = getByText('page not found')
    expect(notFound).toBeInTheDocument()
  })
  test('should render page not found with content type', () => {
    const contentType = 'test content type'
    const { getByText } = render(
      <ContextAndRouterProvider>
        <NotFound contentType={contentType} />
      </ContextAndRouterProvider>
    )

    const notFound = getByText(`${contentType} not found`)
    expect(notFound).toBeInTheDocument()
  })
  test('should render generic page not found if no content type provided', () => {
    const { getByText } = render(
      <ContextAndRouterProvider>
        <NotFound />
      </ContextAndRouterProvider>
    )

    const notFound = getByText('page not found')
    expect(notFound).toBeInTheDocument()
  })
  test('should link to the home page', () => {
    const { container } = render(
      <ContextAndRouterProvider>
        <NotFound />
      </ContextAndRouterProvider>
    )

    const homeLink = container.querySelector('a')
    expect(homeLink).toBeInTheDocument()
    expect(homeLink.getAttribute('href')).toBe('/')
  })
})
