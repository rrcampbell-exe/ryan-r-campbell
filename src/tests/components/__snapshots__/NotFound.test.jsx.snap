// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound > default snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="NotFound"
      >
        <div
          class="about-row section-wrapper"
        >
          <div>
            <svg
              class="about-img"
              id="robot-svg"
              viewBox="0 0 256 256"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M200,48H136V16a8,8,0,0,0-16,0V48H56A32.03635,32.03635,0,0,0,24,80V192a32.03635,32.03635,0,0,0,32,32H200a32.03635,32.03635,0,0,0,32-32V80A32.03635,32.03635,0,0,0,200,48ZM72,108a12,12,0,1,1,12,12A12,12,0,0,1,72,108Zm28,76H92a16,16,0,0,1,0-32h8Zm40,0H116V152h24Zm24,0h-8V152h8a16,16,0,0,1,0,32Zm8-64a12,12,0,1,1,12-12A12,12,0,0,1,172,120Z"
              />
            </svg>
          </div>
          <div
            class="about-description"
          >
            <h2>
              page
               not found
            </h2>
            <p>
              I couldn't find the 
              page
               you were looking for. It might have been moved or relocated as part of a recent site migration.
            </p>
            <a
              href="/"
            >
              return to the homepage here.
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="NotFound"
    >
      <div
        class="about-row section-wrapper"
      >
        <div>
          <svg
            class="about-img"
            id="robot-svg"
            viewBox="0 0 256 256"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M200,48H136V16a8,8,0,0,0-16,0V48H56A32.03635,32.03635,0,0,0,24,80V192a32.03635,32.03635,0,0,0,32,32H200a32.03635,32.03635,0,0,0,32-32V80A32.03635,32.03635,0,0,0,200,48ZM72,108a12,12,0,1,1,12,12A12,12,0,0,1,72,108Zm28,76H92a16,16,0,0,1,0-32h8Zm40,0H116V152h24Zm24,0h-8V152h8a16,16,0,0,1,0,32Zm8-64a12,12,0,1,1,12-12A12,12,0,0,1,172,120Z"
            />
          </svg>
        </div>
        <div
          class="about-description"
        >
          <h2>
            page
             not found
          </h2>
          <p>
            I couldn't find the 
            page
             you were looking for. It might have been moved or relocated as part of a recent site migration.
          </p>
          <a
            href="/"
          >
            return to the homepage here.
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
